controller/ResultController.php

```diff
@@ -0,0 +1,239 @@
+{
+    public const XML_CONTENT_TYPE = 'application/xml';
+
+    /**
+     * @noinspection PhpUnused
+     * @throws common_exception_MethodNotAllowed
+     * @throws tao_models_classes_UserException
+     */
+    public function manageResults()
+    {
+        if (!$this->isRequestPost()) {
+            throw new common_exception_MethodNotAllowed(null, 0, [Request::HTTP_POST]);
+        }
+
+        try {
+            $user = $this->authorizeUser($this->getPsrRequest());
+        } catch (tao_models_classes_UserException $userException) {
+            throw $userException;
+        } catch (Throwable $throwable) {
+            $this->response = $this->createInternalErrorResponse($throwable);
+            return;
+        }
+
+        $payload = (string) $this->getPsrRequest()->getBody();
+        $requestParser = $this->getRequestParser();
+
+        try {
+            $lisRequest = $requestParser->parse($payload);
+            $this->response = $this->processLisRequest(
+                $lisRequest,
+                $user->getLtiProvider()
+            );
+        } catch (ParsingException $parsingException) {
+            $this->response = $this->createParseErrorResponse($parsingException);
+        } catch (Throwable $throwable) {
+            $this->response = $this->createInternalErrorResponse($throwable);
+        }
+    }
+
+    /**
+     * @param ServerRequestInterface $request
+     * @return LtiProviderUser
+     * @throws tao_models_classes_UserException
+     */
+    private function authorizeUser(ServerRequestInterface $request)
+    {
+        try {
+            return $this->getLisAuthAdapterFactory()->create($request)->authenticate();
+        } catch (common_user_auth_AuthFailedException $authFailedException) {
+            throw new tao_models_classes_UserException($authFailedException->getMessage());
+        }
+    }
+
+    /**
+     * @param LisOutcomeRequest $lisRequest
+     * @param LtiProvider $ltiProvider
+     * @return ResponseInterface
+     * @throws common_exception_Error
+     * @throws common_exception_NotFound
+     */
+    private function processLisRequest(LisOutcomeRequest $lisRequest, LtiProvider $ltiProvider)
+    {
+        $lisResponse = $this->getLtiResultService()->process($lisRequest, $ltiProvider);
+
+        $serializer = $this->getOperationsCollection()->getResponseSerializer($lisResponse);
+        if ($serializer === null) {
+            return $this->createInternalErrorResponse();
+        }
+        $xml = $serializer->toXml($lisResponse);
+        $httpStatus = $this->mapLisResponseStatusToHttp($lisResponse->getStatus());
+        return $this->getXmlResponse($httpStatus, $xml);
+    }
+
+    /**
+     * @param string $lisResponseStatus
+     * @return int
+     */
+    private function mapLisResponseStatusToHttp($lisResponseStatus)
+    {
+        switch ($lisResponseStatus) {
+            case LisOutcomeResponseInterface::STATUS_SUCCESS: return StatusCode::HTTP_CREATED;
```

Comment:
Does it complies to  https://www.php-fig.org/psr/psr-2/#52-switch-case ?

Reply:
Is your question about indentation?

Reply:
indentation fixed